File: Navtrack.Core/ClientApp/src/services/api/types/asset/AssetModel.ts
import { LocationModel, DefaultLocationModel } from "../location/LocationModel";

export type AssetModel = {
  id: number;
  name: string;
  deviceId: number;
  deviceType: string;
  location: LocationModel;
};

export const DefaultAssetModel: AssetModel = {
  id: 0,
  name: "",
  deviceId: 0,
  deviceType: "",
  location: DefaultLocationModel
};

File: Navtrack.Api/Models/AssetModel.cs
using System.ComponentModel.DataAnnotations;

namespace Navtrack.Api.Models
{
    public class AssetModel : IModel
    {
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; }
        public int DeviceId { get; set; }
        
        public LocationModel Location { get; set; }
        public DeviceModel Device { get; set; }
    }
}